autoload/ale/references.vim

```diff
@@ -115,7 +122,17 @@ function! s:FindReferences(linter) abort
     \]))
 endfunction
 
-function! ale#references#Find() abort
+function! ale#references#Find(...) abort
+    let s:use_relative_paths = 0
+
+    if len(a:000) > 0
+        for l:option in a:000
+            if l:option == '-relative'
+                let s:use_relative_paths = 1
+            endif
+        endfor
+    endif
+
     for l:linter in ale#linter#Get(&filetype)
         if !empty(l:linter.lsp)
             call s:FindReferences(l:linter)
```

Comment:
Instead of modifying a script variable, pass the options in here. Forward that on to the `OnReady` function. You can then save the options in `s:references_map`. You can write `let l:options = remove(s:references_map, a:response.id)` above to get the options out at the end.

Reply:
Thanks for pointing me in the right direction. I applied your solution and added tests.

Lemme know if there's anything else I can do to clean up (or correct) the code.